wgengine/netstack/netstack_test.go

```diff
@@ -74,3 +77,52 @@ func getMemStats() (ms runtime.MemStats) {
 	runtime.ReadMemStats(&ms)
 	return
 }
+
+func TestRemoveECNBits(t *testing.T) {
+	// withECNHex is a TCP SYN packet with ECN bits set in the TCP
+	// header as captured by Wireshark on macOS against the
+	// Tailscale interface.  For whatever reason, the ECN bits are
```

Comment:
"Since the Transmission Control Protocol (TCP) does not perform congestion control on control packets (pure ACKs, SYN, FIN segments), control packets are usually not marked as ECN-capable."

That's why :) You captured a SYN, which indicates its ECN-awareness in the TCP headers, but by design does not advertise ECN-awareness at the IP layer.

Reply:
Thanks, updated comment.